cypress/tests/data-source.spec.ts

```diff
@@ -0,0 +1,52 @@
+import { DATA_FEDERATION_NAMESPACE } from '../constants/common';
+import { TEST_DATA_SOURCE } from '../constants/tests';
+import {
+  checkDataSourceCreation,
+  createDataSource,
+  deleteDataSourceResources,
+  deleteUsingDetailsPageKebabMenu,
+  navigateToCreatePage,
+  Providers,
+} from '../views/data-resource';
+import { MCGMSCommon } from '../views/mcg-ms-common';
+
+describe('tests for data source creation', () => {
+  before(() => {
+    cy.login();
+    cy.visit('/');
+  });
+
+  beforeEach(() => {
+    MCGMSCommon.visitDataSourceListPage();
+    navigateToCreatePage();
+  });
+
+  afterEach(() => {
+    deleteUsingDetailsPageKebabMenu(TEST_DATA_SOURCE);
+    deleteDataSourceResources(TEST_DATA_SOURCE, DATA_FEDERATION_NAMESPACE);
```

Comment:
I see the creation tests but for deletion there is no single test but this under-the-hood deletion/cleanup.
Is it possible to create a test that verifies the deletion from the GUI (at least the first time)?
The benefit is increasing the test coverage and deletion being explicit over implicit.
That said, I'm not against this shortcut if needed after every test, just pointing out that the GUI should be able to delete resources (at least the ones related to the addon).

Reply:
agreed !!
If u are creating data source from the UI, u should delete it from UI as well...

Reply:
sure adding in the tests for the data source deletion also